app.rb

```diff
@@ -2,123 +2,166 @@
+      handle_data(people_data, 'student')
     when 2
-      print 'Specialisation?'
+      puts 'Specialisation: '
       specialisation = gets.chomp
-      people.push(Teacher.new(specialisation, age, name))
+      @person << people_data = Teacher.new(specialisation, age, name)
+      handle_data(people_data, 'teacher')
     end
     puts 'Person added successfully'
   end
 
-  def add_book(books)
+  def add_book
     puts 'Please Add a book'
-    print 'Title :'
+    puts 'Title :'
     title = gets.chomp.capitalize
-    print 'Author:'
+    puts 'Author:'
     author = gets.chomp.capitalize
-    books.push(Book.new(title, author))
+    @book << Book.new(title, author)
+    book_info = Book.new(title, author)
+    book_data_hash = { title: book_info.title, author: book_info.author }
+    @book_data.push(book_data_hash)
     puts 'Book added successfully'
   end
 
-  def add_rental(books, people, rental_data)
-    puts 'Please select abook from the list by number'
-    books.map.with_index { |book, index| puts "#{index} Title: #{book.title}',Auther:#{book.author}" }
-    selected_book = gets.chomp.to_i
-    puts "Choose a person from the list:(
-      DON'T CHOOSE ID PLEASE)"
-    people.map.with_index do |person, index|
-      puts "#{index}, Name: #{person.name} Age: #{person.age},
-          ID:#{person.id}"
+  def add_rental
+    puts 'Please select a book from the list by number'
+    @book.map.with_index { |item, index| puts "#{index} Title: #{item.title}',Auther:#{item.author}" }
+    selected_book_id = gets.chomp.to_i
+    selected_book = @book[selected_book_id]
+    puts "Choose a person from the list:(DON'T CHOOSE ID PLEASE)"
+    @person.map.with_index do |item, index|
+      puts "#{index}, Name: #{item.name} Age: #{item.age}, ID:#{item.id}"
     end
-    selected_person = gets.chomp.to_i
+    selected_person_id = gets.chomp.to_i
+    selected_person = @person[selected_person_id]
 
-    print 'date?'
-    selecteted_date = gets.chomp
-    rental_data.push(Rental.new(selecteted_date, people[selected_person], books[selected_book]))
-    puts 'rental_data updated'
+    puts 'Date?'
+    selected_date = gets.chomp
+    @rental << Rental.new(selected_date, selected_person, selected_book)
+    # rental_data_hash = { date: selected_date, book_index: selected_book, person_index: selected_person }
+    # @rental_data.push(rental_data_hash)
+    puts 'Rental created sucessfully!'
   end
 
-  def show_rental_data(rentals)
-    print 'Person ID'
+  def show_rental
+    puts 'Person ID'
     selected_person_id = gets.chomp.to_i
-    rentals.each do |rental|
-      next unless rental.person.id == selected_person_id
+    @person.each do |item|
+      next unless item.id == selected_person_id
 
-      puts rental.book.title
-      puts rental.person.name
-      puts rental.date
+      item.rental_data.each do |rental|
+        puts rental.book.title
+        puts rental.person.name
+        puts rental.date
+      end
```

Comment:
- [ ] Great work with the refactor. I also noticed that when I create a rental and exit the application, I can't see the rental when I start the app again. I would suggest you do the following: 
- save the people with their ID correctly in the JSON file
- When saving the rentals, save the ID of the person renting it as well. (this is after the first step)
- Then access the data based on the ID of the person that is correctly saved. 👍